config/default.json

```diff
@@ -6,7 +6,6 @@
   "assetPath": "assets",
   "jsSourcePath": "src/js",
   "s3": {
-    "s3Deploy": false,
     "bucket": "if-any",
     "defaultCDNBase": "if-any"
```

Comment:
It is used in here https://github.com/iroy2000/react-redux-boilerplate/blob/master/webpack.config.build.babel.js#L16

Reply:
Right, I guess what I meant was: if we use an `S3_DEPLOY` env var, either through `dotenv` or through a custom NPM script (as in the PR), would it make sense to remove this block from the config?  

It doesn't appear that the `s3.defaultCDNBase` or `s3.bucket` value are used anywhere in the project currently, but rather environment variables, like so:

```javascript
    s3UploadOptions: {
      Bucket: process.env.AWS_BUCKET,
    },
    cdnizerCss: {
      test: /images/,
      cdnUrl: process.env.AWS_CDN_URL,
    },
```

I'm suggesting that, since the project already uses environment variables for 2 of the three s3 values, we just do away with the s3 block in the config and just use env variables.  It seems like it would be less confusing, but maybe I'm missing something.

Reply:
I think that make sense as well, need to make sure we document those.

The initial thought was having all config consolidated inside a config file to make it very explicit. 

But I do see your argument.

Reply:
@iroy2000 In working on this, I did not anticipate that users might want to make an S3 deploy using different config files.

I don't want my change to break that functionality, but: right now, I'm not sure if this actually works.  As I mentioned above, the AWS_BUCKET and AWS_CDN_URL are actually in environment variables, not read from the configs.  

It appears to me that the only thing you can `config` per environment is `s3Deploy`, and the other values are all read from the same `.env` file.

Am I correct about this?

Reply:
Re: `I did not anticipate that users might want to make an S3 deploy using different config files`. It is normal though user will have at least `dev`, `prod` account and it might be different. As long as this use case is not broken, it is fine.   

Re: `If so, maybe we recommend env files per stage, like test.env, production.env`
`.env` file initially is for development purpose. Because `.env` file might contain credential and that should not be part of the committed source code. 

And the original use case of this boilerplate is through Jenkins, which you can inject node variables during build time. 

But even when I said `.env` should not be part of the source code, it could still be  a step in your deployment system to `generate or pull in` a `.env` file.  So instead of check in `test.env` or `prod.env` into the same repo, your file is always called `.env`.

This needs to communicate well to the user `why` we made that choice.